### PR TITLE
Split scipp-developer.yml into platform-specific files

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
This allows us to specify dependencies like `gxx_linux` on linux platforms while not breaking windows/osx conda environments.

### CODE REVIEW

File: scipp-developer-windows.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - pre-commit
+  - yapf
+  - flake8
+
+  # Docs
+  - pandoc
+  - sphinx
+  - sphinx_rtd_theme
+  - nbsphinx
+  - docutils=0.16 # see https://github.com/spatialaudio/nbsphinx/issues/549
```
Comment (OwenArnold) [inline scipp-developer-windows.yml:48]: Did you consider using any tools like this https://conda-devenv.readthedocs.io/en/latest/usage.html ?

Comment (OwenArnold) [reply scipp-developer-windows.yml:48]: I like the idea of not maintaining almost identical environment files, but dislike having to use more tooling. I'm honestly not sure which is going to be worse for us.

Comment (SimonHeybrock) [reply scipp-developer-windows.yml:48]: How would we install `conda-devenv`?

Comment (Tom-Willemsen) [reply scipp-developer-windows.yml:48]: It would have to be manually installed by developers, I guess into the base conda environment.

**Comment by SimonHeybrock** [comment]
Closing, since @nvaytet will likely apply the equivalent of https://github.com/scipp/scippneutron/pull/259 here.

**Pull request closed without merging**